scopes/dependencies/pnpm/lockfile-deps-graph-converter.ts

```diff
@@ -244,6 +268,38 @@ export function convertGraphToLockfile(
       }
     }
   }
+  const pkgsToResolve: Array<{ name: string; version: string; pkgId: string }> = [];
+  const pkgsInTheWorkspaces = new Set(Object.values(manifests).map(({ name }) => name));
+  for (const [pkgId, pkg] of Object.entries(lockfile.packages)) {
+    if (pkg.resolution == null || 'type' in pkg.resolution && pkg.resolution.type === 'directory') {
```

Comment:
you can probably replace `'type' in pkg.resolution && pkg.resolution.type === 'directory'` with `pkg.resolution?.type === 'directory'`

Reply:
I can't because it is a union type and not every type has the type field.

![image](https://github.com/user-attachments/assets/f84d4d6c-9576-41f0-b538-ee3193145e02)

Reply:
Makes sense.